examples/mobile/HomeAppServer/app.js

```diff
@@ -1,64 +1,160 @@
 const homeAppPath = "../HomeApp";
+const WebSocket = require("ws");
 
 var express = require("express");
 var path = require("path");
 var cookieParser = require("cookie-parser");
 var logger = require("morgan"),
-	app = express();
+	app = express(),
+	ws,
+	pingCount = 0,
+	result = {},
+	apps = [{
+		"appID": "vUf39tzQ3s.UIComponents",
+		"isInstalled": true,
+		"isActive": true,
+		"webClipsList": [
+			{
+				url: "webclip/apps-on-tv"
+			},
+			{
+				url: "webclip/latest-news"
+			}
+		]
+	},
+	{
+		"appID": "vUf39tzQ3t.UIComponents",
+		"isInstalled": true,
+		"isActive": false,
+		"webClipsList": [
+			{
+				url: "webclip/now-on-tv"
+			},
+			{
+				url: "webclip/restaurant"
+			}
+		]
+	},
+	{
+		"appID": "vUf39tzQ3r.UIComponents",
+		"isInstalled": false,
+		"isActive": false,
+		"webClipsList": [
+			{
+				url: "webclip/tv-remote-control"
+			},
+			{
+				url: "webclip/weather"
+			}
+		]
+	}],
+	apps2 = [{
+		"appID": "TUf39tzQ3s.UIComponents",
+		"isInstalled": true,
+		"isActive": true,
+		"webClipsList": [
+			{
+				url: "webclip/apps-on-tv"
+			},
+			{
+				url: "webclip/latest-news"
+			}
+		]
+	},
+	{
+		"appID": "vUf39tzQ3t.UIComponents",
+		"isInstalled": true,
+		"isActive": false,
+		"webClipsList": [
+			{
+				url: "webclip/now-on-tv"
+			},
+			{
+				url: "webclip/restaurant"
+			}
+		]
+	},
+	{
+		"appID": "vUf39tzQ3r.UIComponents",
+		"isInstalled": false,
+		"isActive": false,
+		"webClipsList": [
+			{
+				url: "webclip/tv-remote-control"
+			},
+			{
+				url: "webclip/weather"
+			}
+		]
+	}];
 
 app.use(logger("dev"));
 app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
 app.use(cookieParser());
```

Comment:
do we use cookies?

Reply:
I don't know. This is your code.

Reply:
We can left cookie. Cookies are useful for user session